pennylane_lightning/core/src/measurements/tests/Test_MeasurementsBase.cpp

```diff
@@ -66,31 +81,42 @@ template <typename TypeList> void testProbabilities() {
-                       0.0870997, 0.00397696, 0.02280642, 0.00104134}},
-                     {{1, 2, 0},
-                      {0.67078706, 0.0870997, 0.17564072, 0.02280642,
-                       0.03062806, 0.00397696, 0.00801973, 0.00104134}},
-                     {{2, 0, 1},
-                      {0.67078706, 0.17564072, 0.03062806, 0.00801973,
-                       0.0870997, 0.02280642, 0.00397696, 0.00104134}},
-                     {{2, 1, 0},
-                      {0.67078706, 0.17564072, 0.0870997, 0.02280642,
-                       0.03062806, 0.00801973, 0.00397696, 0.00104134}},
-                     {{0, 1}, {0.70141512, 0.09107666, 0.18366045, 0.02384776}},
-                     {{0, 2}, {0.75788676, 0.03460502, 0.19844714, 0.00906107}},
-                     {{1, 2}, {0.84642778, 0.0386478, 0.10990612, 0.0050183}},
-                     {{2, 1}, {0.84642778, 0.10990612, 0.0386478, 0.0050183}},
-                     {{0}, {0.79249179, 0.20750821}},
-                     {{1}, {0.88507558, 0.11492442}},
-                     {{2}, {0.9563339, 0.0436661}}};
+            input = {
+#ifdef _ENABLE_PLGPU
+                // Bit index reodering conducted in the python layer
+                // for L-GPU. Also L-GPU backend doesn't support
+                // out of order wires for probability calculation
+                {{2, 1, 0},
+                 {0.67078706, 0.03062806, 0.0870997, 0.00397696, 0.17564072,
+                  0.00801973, 0.02280642, 0.00104134}}
+#else
+                {{0, 1, 2},
+                 {0.67078706, 0.03062806, 0.0870997, 0.00397696, 0.17564072,
+                  0.00801973, 0.02280642, 0.00104134}},
+                {{0, 2, 1},
+                 {0.67078706, 0.0870997, 0.03062806, 0.00397696, 0.17564072,
+                  0.02280642, 0.00801973, 0.00104134}},
+                {{1, 0, 2},
+                 {0.67078706, 0.03062806, 0.17564072, 0.00801973, 0.0870997,
+                  0.00397696, 0.02280642, 0.00104134}},
+                {{1, 2, 0},
+                 {0.67078706, 0.0870997, 0.17564072, 0.02280642, 0.03062806,
+                  0.00397696, 0.00801973, 0.00104134}},
+                {{2, 0, 1},
+                 {0.67078706, 0.17564072, 0.03062806, 0.00801973, 0.0870997,
+                  0.02280642, 0.00397696, 0.00104134}},
+                {{2, 1, 0},
+                 {0.67078706, 0.17564072, 0.0870997, 0.02280642, 0.03062806,
+                  0.00801973, 0.00397696, 0.00104134}},
+                {{0, 1}, {0.70141512, 0.09107666, 0.18366045, 0.02384776}},
+                {{0, 2}, {0.75788676, 0.03460502, 0.19844714, 0.00906107}},
+                {{1, 2}, {0.84642778, 0.0386478, 0.10990612, 0.0050183}},
+                {{2, 1}, {0.84642778, 0.10990612, 0.0386478, 0.0050183}},
+                {{0}, {0.79249179, 0.20750821}},
+                {{1}, {0.88507558, 0.11492442}},
+                {{2}, {0.9563339, 0.0436661}}
```

Comment:
Wait, so all the other backends support out of order indices here?

Reply:
Correct, other backends have transpose support.